src/app/components/Rewards.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  rewardSlice,
+  fetchReciepts,
+  fetchAccountRewards,
+  fetchOrderRewards,
+} from "../state/rewardSlice";
+import { useEffect } from "react";
+import { useAppDispatch, useAppSelector } from "../hooks";
+import { useSelector } from "react-redux";
+import type { RootState } from "../state/store";
+
+function ClaimButton() {
+  const dispatch = useAppDispatch();
+  const handleClaim = async () => {
+    await dispatch(fetchReciepts()); //Get all users reciepts already exists
+    dispatch(rewardSlice.actions.claimRewards());
+  };
+
+  return (
+    <button
+      className="btn btn-wide btn-primary-content btn-accent uppercase"
+      onClick={() => {
+        handleClaim();
+      }}
+    >
+      Claims
+    </button>
+  );
+}
+
+function ClaimButton2() {
+  const dispatch = useAppDispatch();
+  const handleClaim = async () => {
+    await dispatch(fetchReciepts());
+    await dispatch(fetchAccountRewards());
+    await dispatch(fetchOrderRewards());
+  };
+
+  return (
+    <button
+      className="btn btn-wide btn-primary-content btn-accent uppercase"
+      onClick={() => {
+        handleClaim();
+      }}
+    >
+      Load Rewards
+    </button>
+  );
+}
+
+function TotalEarned() {
+  const dispatch = useAppDispatch();
+  const rewardData = useSelector(
+    (state: RootState) => state.rewardSlice.rewardData
+  );
+
+  useEffect(() => {
+    async function fetchData() {
+      await dispatch(fetchReciepts());
+      await dispatch(fetchAccountRewards());
+      await dispatch(fetchOrderRewards());
+    }
+    fetchData();
+  }, []);
```

Comment:
missing dependency, please configure your IDE to show eslint warnings/errors during development